controller/registries/consul/config.go

```diff
@@ -159,15 +180,71 @@ func (reg *Consul) Stop() error {
 
 	reg.lock.Lock()
 	defer reg.lock.Unlock()
+	for key := range reg.softDeletedServices {
+		if _, ok := reg.watchingServices[key]; !ok {
+			reg.store.Delete(reg.getServiceEntryKey(key.Tag, key.ServiceName))
+		}
+	}
+	for key := range reg.watchingServices {
+		reg.removeService(key)
+	}
 
 	return nil
 }
 
 func (reg *Consul) Reload(c registrytype.RegistryConfig) error {
-	fmt.Println(c)
+	config := c.(*consul.Config)
+
+	client, err := reg.NewClient(config)
+	if err != nil {
+		return err
+	}
+
+	reg.client = client
+
+	fetchedServices, err := reg.fetchAllServices(client)
+	if err != nil {
+		return fmt.Errorf("fetch all services error: %v", err)
+	}
+
+	for key := range reg.softDeletedServices {
+		if _, ok := fetchedServices[key]; !ok {
+			reg.store.Delete(reg.getServiceEntryKey(key.Tag, key.ServiceName))
+		}
+	}
+	reg.softDeletedServices = map[consulService]bool{}
+
+	for key := range reg.watchingServices {
+		// unsubscribe with the previous client
```

Comment:
Should move `reg.client = client` after this. Otherwise, it will be unsubscribed with the current client.